api/v1alpha1/nodepool_types.go

```diff
@@ -553,6 +553,12 @@ type AWSNodePoolPlatform struct {
 	// +kubebuilder:validation:MaxItems=25
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// NodepoolArch is an optional parameter specifying the preferred processor
+	// architecture for the nodepool
+	//
+	// +optional
+	NodepoolArch string `json:"nodepoolarch"`
```

Comment:
Why is this an aws-specific setting?

Reply:
@alvaroaleman ATM, AWS will be the only place were we'll support arm workers.

Reply:
hmm, maybe we should define nodepoolarch in `type NodePoolSpec struct` [here](https://github.com/openshift/hypershift/blob/786985d2e4291556154ea27385d856e8099b6c73/api/v1alpha1/nodepool_types.go#L78) 
@alvaroaleman is that what you were suggesting?

Reply:
Yeah, that is what I meant. API-wise it belongs there IMHO. Even if that is not going to work immediately on all platforms, the goal is still to get it supported eveywhere, right?